example/echo_event_loop.ml

```diff
@@ -0,0 +1,172 @@
+open Core
+open Core.Unix
+
+module Ring = struct
+  type t = { ring : (int -> int -> unit) Io_uring.t }
+
+  let make ~ring_size =
+    { ring =
+        Io_uring.create
+          ~max_submission_entries:ring_size
+          ~max_completion_entries:(ring_size * 2)
+    }
+  ;;
+
+  let wait t =
+    let _submitted = Io_uring.submit t.ring in
+    Stdio.Out_channel.flush Stdio.Out_channel.stdout;
+    Io_uring.wait t.ring ~timeout:`Never;
+    Io_uring.iter_completions t.ring ~f:(fun ~user_data ~res ~flags ->
+        user_data res flags);
+    Io_uring.clear_completions t.ring;
+    ()
+  ;;
+
+  let global = make ~ring_size:1024
+end
+
+module Deferred = struct
+  type 'a t =
```

Comment:
I recommend using Monad.Make to get nice functor, applicative, and monad functions with Let_syntax support.